hack/pipeline-validator/main.go

```diff
@@ -0,0 +1,147 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+// This tool is using to validate Jenkins pipelines.
+// Based on https://jenkins.io/doc/book/pipeline/development/#linter
+
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"mime/multipart"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+const JenkinsURL string = "https://devops-ci.elastic.co"
+
+var client = http.Client{Timeout: 60 * time.Second}
+
+type CSRFToken struct {
+	Crumb             string `json:"crumb"`
+	CrumbRequestField string `json:"crumbRequestField"`
+}
+
+func main() {
+	log.SetFlags(log.LstdFlags | log.Lshortfile)
+
+	fmt.Println("Starting validation of Jenkins pipelines...")
+
+	pipelines, err := listPipelines()
+	if err != nil {
+		log.Fatalln("Can't find pipelines!", err)
+	}
+
+	fmt.Println("Getting Jenkins CSRF token...")
+	token, err := getToken()
+	if err != nil {
+		log.Fatalln("Can't get CSRF token!", err)
+	}
+
+	for _, v := range pipelines {
+		_, err := validate(token, v)
+		fmt.Println("Validating", v)
+		if err != nil {
+			log.Println("Error during validation of:", v)
+			log.Fatalln(err)
+		}
+	}
+
+	fmt.Println("Validation was successful!")
+}
+
+func listPipelines() ([]string, error) {
+	var files []string
+	err := filepath.Walk(".", func(path string, info os.FileInfo, err error) error {
+		if !info.IsDir() {
+			if strings.Contains(info.Name(), "Jenkinsfile") || strings.Contains(info.Name(), ".jenkinsfile") {
+				files = append(files, path)
+			}
+		}
+		return nil
+	})
+	return files, err
+}
+
+func getToken() (*CSRFToken, error) {
+	req, err := http.NewRequest("GET", JenkinsURL+"/crumbIssuer/api/json", http.NoBody)
```

Comment:
Nit: It's a good habit to use `filepath.Join(`. It can avoid to end up with some `//`.

Reply:
Fixed